src/reducers/authenticationReducer.js

```diff
@@ -16,22 +21,32 @@ import {
   FLAG_SETTING_GEOLOCATION_ENABLED,
   FLAG_GETTING_GEOLOCATION_ENABLED,
 } from '../types';
+import { updateActivityList } from './common';
 import {
+  AUTHENTICATED,
   DEFAULT_DEVELOPER_MODE,
   DEFAULT_LANGUAGE,
   DEFAULT_GEOLOCATION_ENABLED,
 } from '../config/constants';
-import { updateActivityList } from './common';
 
-const INITIAL_STATE = Map({
-  current: Map({
-    geolocation: Map(),
-    folder: null,
+export const DEFAULT_USER_SETTINGS = {
+  lang: DEFAULT_LANGUAGE,
+  developerMode: DEFAULT_DEVELOPER_MODE,
+  geolocationEnabled: DEFAULT_GEOLOCATION_ENABLED,
+};
+
+export const DEFAULT_USER = {
+  geolocation: Map(),
+  settings: { ...DEFAULT_USER_SETTINGS },
+};
+
+export const INITIAL_STATE = Map({
+  current: {
     activity: List(),
-    lang: DEFAULT_LANGUAGE,
-    developerMode: DEFAULT_DEVELOPER_MODE,
-    geolocationEnabled: DEFAULT_GEOLOCATION_ENABLED,
-  }),
+    folder: null,
+  },
+  authenticated: false,
+  user: Map(DEFAULT_USER),
```

Comment:
I saw somewhere else that the `id` for the `user` is inside the user as `userId`. I would expect this to be inside the `user` object as `id`. Something like `userId = user.id`. What do you think?